lib/src/main/java/uk/gov/di/ipv/cri/kbv/api/security/SoapTokenRetriever.java

```diff
@@ -34,7 +43,49 @@ public String getSoapToken() {
             }
             retry++;
         }
-        return token;
+
+        if (!validTokenFetch && !isCachedTokenValid()) {
+            try {
+                if (token != null
+                        && SoapTokenUtils.isTokenValid(SoapTokenUtils.decodeTokenPayload(token))) {
+                    cachedToken = token;
+                    LOGGER.info(
+                            "Updated cached token with the one received from Experian. The token given by Experian is valid but not within our threshold, using anyway...");
+                }
+            } catch (Exception ignore) {
+                LOGGER.warn("Cached SOAP token and token from Experian are both invalid");
+            }
+
+            return token;
+        }
+
+        if (validTokenFetch) {
+            LOGGER.info("Cached token has been updated");
+            cachedToken = token;
+        } else {
+            LOGGER.warn(
+                    "Received an invalid SOAP token from Experian after retries, using valid cached token for now...");
+        }
+
+        return cachedToken;
+    }
+
+    public boolean isCachedTokenValid() {
+        try {
+            return cachedToken != null
+                    && SoapTokenUtils.isTokenValid(SoapTokenUtils.decodeTokenPayload(cachedToken));
+        } catch (Exception ignored) {
+            return false;
+        }
+    }
+
+    public boolean isCachedTokenValidAndWithinThreshold() {
+        try {
+            return cachedToken != null
+                    && isTokenValid(SoapTokenUtils.decodeTokenPayload(cachedToken));
+        } catch (Exception ignored) {
+            return false;
+        }
```

Comment:
same here

Reply:
It'll be a JacksonProcessingException or IndexOutOfBounds - if we log this, we'll see an error message every cold start.